python/taichi/main.py

```diff
@@ -88,6 +92,48 @@ def __call__(self):
 
         return getattr(self, args.command)(sys.argv[2:])
 
+    @staticmethod
+    def _check_version():
+        # Check Taichi version for the user.
+        print('Checking your Taichi version...')
+        major = _ti_core.get_version_major()
+        minor = _ti_core.get_version_minor()
+        patch = _ti_core.get_version_patch()
+        version = f'{major}.{minor}.{patch}'
+        payload = {'version': version, 'platform': '', 'python': ''}
+
+        system = platform.system()
+        if system == 'Linux':
+            payload['platform'] = 'manylinux1_x86_64'
+        elif system == 'Windows':
+            payload['platform'] = 'win_amd64'
+        elif system == 'Darwin':
+            if platform.release() < '19.0.0':
+                payload['platform'] = 'macosx_10_14_x86_64'
+            elif platform.machine() == 'x86_64':
+                payload['platform'] = 'macosx_10_15_x86_64'
+            else:
+                payload['platform'] = 'macosx_11_0_arm64'
+
+        python_version = platform.python_version()
+        if python_version.startswith('3.6'):
+            payload['python'] = 'cp36'
+        elif python_version.startswith('3.7'):
+            payload['python'] = 'cp37'
+        elif python_version.startswith('3.8'):
+            payload['python'] = 'cp38'
+        elif python_version.startswith('3.9'):
+            payload['python'] = 'cp39'
+
+        response = requests.post('http://54.90.48.192/check_version',
+                                 json=payload).json()
+        if response['status'] == 1:
+            print(
+                f'Your Taichi version {version} is outdated. The latest version is {response["latest_version"]}, you can use pip to upgrade to the latest Taichi!'
+            )
+        elif response['status'] == 0:
+            print(response['message'])
+
```

Comment:
Several unexpected things could happen during an HTTP call, it can timeout, the response might not be JSON-serializable, etc... I'd probably:

- add a timeout to the POST method (which might raise an exception)
- store the response as is and use something like `response.raise_for_status()` before trying to parse the response into JSON. (which might raise a different type of exception)
- Wrap the caller of this static method in a try catch finally block to catch the above exceptions so even if the HTTP call failed, user won't get exposed to some bizarre issues due to Network failure.

WDYT?

Reply:
optional nit: we may want to add the link to our docs or print the explicit pip install command directly in the helper message for users.